pkg/destroy/azure/azure.go

```diff
@@ -492,6 +486,23 @@ func wasNotFound(resp *http.Response) bool {
 	return resp != nil && resp.StatusCode == http.StatusNotFound
 }
 
+func isNotFoundError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	var dErr autorest.DetailedError
+	if errors.As(err, &dErr) {
+		switch statusCode := dErr.StatusCode.(type) {
+		case int:
+			if statusCode == http.StatusNotFound {
+				return true
+			}
+		}
```

Comment:
This is overkill for an equality check. I don't know why the `isResourceGroupBlockedError` function does this. But let's keep it as is for consistency. We can change them both as a separate PR if we like.
```suggestion
		return dErr.StatusCode == http.StatusNotFound
```

Reply:
Yep. I thought about it but kept it to make it look consistent with the other fns. In any case, I'll update. Please lgtm again. Thanks

Reply:
Doh. Got merged before I can make this change. Will create a separate PR to simplify these.